trestlebot/tasks/authored/compdef.py

```diff
@@ -158,6 +169,63 @@ def create_new_default(
         )
         rules_view_builder.write_to_yaml(rule_dir)
 
+    def create_update_cac_compdef(
+        self,
+        comp_description: str,
+        comp_type: str,
+        product: str,
+        cac_content_root: str,
+        working_dir: str,
+    ) -> None:
+        """Create component definition for cac content
+
+        Args:
+            comp_description: Description of the component
+            comp_type: Type of the component
+            product: Product name for the component
+            cac_content_root: ComplianceAsCode repo path
+            working_dir: workplace repo path
+        """
+        # Initial component definition fields
+        component_definition = generate_sample_model(ComponentDefinition)
+        component_definition.metadata.title = f"Component definition for {product}"
+        component_definition.metadata.version = "1.0"
+        component_definition.components = list()
+        oscal_component = generate_sample_model(DefinedComponent)
+        oscal_component.title = get_component_title(product, cac_content_root)
+        oscal_component.type = comp_type
+        oscal_component.description = comp_description
+
+        # Create all of the component properties for rules
+        # This part will be updated in CPLYTM-218
+        """
+        rules: List[RuleInfo] = self.rules_transformer.get_all_rules()
+        all_rule_properties: List[Property] = self.rules_transformer.transform(rules)
+        oscal_component.props = none_if_empty(all_rule_properties)
+        """
+        repo_path = pathlib.Path(working_dir)
+        out_path = repo_path.joinpath(f"{const.MODEL_DIR_COMPDEF}/{product}/")
+        oname = "component-definition.json"
+        ofile = out_path / oname
+        if ofile.exists():
+            logger.info(f"The component for product {product} exists.")
+            with open(ofile, "r", encoding="utf-8") as f:
+                data = json.load(f)
+            for component in data["component-definition"]["components"]:
+                if component.get("title") == get_component_title(
+                    product, cac_content_root
+                ):
+                    logger.info("Update the exsisting component definition.")
+                    # Need to update props parts if the rules updated
+                    # Update the version and last modify time
+                    update_component_definition(ofile)
```

Comment:
I assume this function is here without conditions just temporarily, correct? Or is it necessary to update version and timestamp whenever the command is called?

Reply:
Yes, I think it should update the rules/parameters part according to the specific requirement. At that moment, I am not sure how to update that part, but the `version` and `last-modified` should be updated together with all the updates.